apps/server/src/modules/oauth-provider/uc/oauth-provider.consent-flow.uc.spec.ts

```diff
@@ -0,0 +1,182 @@
+import { OauthProviderConsentFlowUc } from '@src/modules/oauth-provider/uc/oauth-provider.consent-flow.uc';
+import { ICurrentUser } from '@shared/domain';
+import { ForbiddenException } from '@nestjs/common';
+import { IdTokenService } from '@src/modules/oauth-provider/service/id-token.service';
+import { IdToken } from '@src/modules/oauth-provider/interface/id-token';
+import resetAllMocks = jest.resetAllMocks;
+import clearAllMocks = jest.clearAllMocks;
+
+describe('OauthProviderConsentFlowUc', () => {
+	let module: TestingModule;
+	let uc: OauthProviderConsentFlowUc;
+	let service: DeepMocked<OauthProviderService>;
+	let idTokenService: DeepMocked<IdTokenService>;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				OauthProviderConsentFlowUc,
+				{
+					provide: OauthProviderService,
+					useValue: createMock<OauthProviderService>(),
+				},
+				{
+					provide: IdTokenService,
+					useValue: createMock<IdTokenService>(),
+				},
+			],
+		}).compile();
+
+		uc = module.get(OauthProviderConsentFlowUc);
+		service = module.get(OauthProviderService);
+		idTokenService = module.get(IdTokenService);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('consent', () => {
+		let challenge: string;
+		let currentUser: ICurrentUser;
+		let consentResponse: ProviderConsentResponse;
+
+		beforeEach(() => {
+			challenge = 'challengexyz';
+			currentUser = { userId: 'userId' } as ICurrentUser;
+			consentResponse = {
+				challenge,
+				subject: currentUser.userId,
+			};
+		});
+
+		afterEach(() => {
+			resetAllMocks();
+		});
+
+		describe('getConsentRequest', () => {
+			it('should call service', async () => {
+				// Act
+				await uc.getConsentRequest(challenge);
+
+				// Assert
+				expect(service.getConsentRequest).toHaveBeenCalledWith(challenge);
+			});
+		});
+
+		describe('patchConsentRequest', () => {
+			let requestBody: ConsentRequestBody | RejectRequestBody;
+			let acceptQuery: AcceptQuery;
+
+			beforeEach(() => {
+				requestBody = {
+					grant_scope: ['openid', 'offline'],
+					remember: false,
+					remember_for: 0,
+				};
+				acceptQuery = { accept: true };
+				consentResponse.requested_scope = requestBody.grant_scope;
+				service.getConsentRequest.mockResolvedValue(consentResponse);
+			});
+
+			afterEach(() => {
+				clearAllMocks();
+			});
+
+			it('acceptConsentRequest: validateSubject should fail and throws forbidden', async () => {
+				// Arrange
+				consentResponse.subject = 'notValidSubject';
+
+				// Act
+				await expect(uc.patchConsentRequest(challenge, acceptQuery, requestBody, currentUser)).rejects.toThrow(
+					ForbiddenException
+				);
+
+				// Assert
+				expect(service.getConsentRequest).toHaveBeenCalledWith(challenge);
+				expect(service.acceptConsentRequest).not.toHaveBeenCalledWith();
+				expect(service.rejectConsentRequest).not.toHaveBeenCalled();
+			});
+
+			it('acceptConsentRequest: should call service', async () => {
```

Comment:
looks like it would make sense to group these testcases under a `describe('acceptConsentRequest')` or better `describe('when consent is accepted')`

Reply:
Done